File: 01_Programming_Basics/01_Programming_Basics_with_C++/08_While_Loop_Lab/02_Graduation.cpp
#include <iostream>
#include <iomanip>
using namespace std;

int main() {
  string name = "";
  double grade = 0;
  double sumGrade = 0;
  double counter = 0;

  cin >> name;
  
  while (true) {
    if (counter == 12) { break; }  

    cin >> grade;

    if (grade >= 4) {
      sumGrade += grade;
      counter++;
    }

  }

  cout << name << " graduated. Average grade: " << fixed << setprecision(2) << sumGrade / 12 << endl; 

  return 0;
  
}

File: 01_Programming_Basics/03_Programming_Basics_with_JavaScript/05_While_Loop/02.Graduation.js
function Graduation(args) {
  let name = args[0];
  let i = 1;
  let n = parseFloat(args[i]);
  let nTotal = 0.0;
  let year = 1;

  while(year <= 12) {

    if (n >= 4.00) {
      nTotal += n;
      year += 1;
    } 

    i += 1;
    n = parseFloat(args[i]);
  }
  
  let average = nTotal / 12;
  console.log(name + " graduated. Average grade: " + average.toFixed(2));
}